r/R/zzz.R

```diff
@@ -18,3 +18,11 @@
 #' @useDynLib arrow, .registration = TRUE
 #' @importFrom Rcpp sourceCpp
 NULL
+
+#' Is the C++ Arrow library available
+#'
+#' @export
```

Comment:
This function could use some more substantive docs, like to say that package users generally wouldn't need to call this, to address the concern that JJ raised about the examples.

Reply:
Yes, we use `\dontrun` extensively for TensorFlow, Spark, and reticulate. We do however have example functions for any code that doesn't have dependencies (in this case e.g. `arrow_available()`) so that we don't trip the "no running examples" flag.

pkgdown is actually setup to deal with this pretty gracefully (example code is enclosed in a subtle not run treatment, e.g. see https://keras.rstudio.com/reference/keras_model.html)

Reply:
Right, pkgdown handles `dontrun` gracefully, but the downside is that you don't get to see the example results, like with e.g. https://dplyr.tidyverse.org/reference/select.html#examples. Provided that we build the pkgdown static site in an environment where libarrow is installed (which we will), if we used the `arrow_available()` check in the examples, we'd get the example output showing. For pedagogical purposes, I'm not sure which is the better tradeoff (i.e. see real output but possibly be mislead that you always have to check if arrow is available, vs. no output but also no extraneous code).

Reply:
Looks like pkgdown has a `run_dont_run` argument (https://github.com/r-lib/pkgdown/blob/89af5341ef2a0eedf07770d3e98f5094dacb9029/R/build-reference.R#L86), so we can have the cleaner `\dontrun` and still get example output on the pkgdown site.

Reply:
Sure. I'd like to document this function in a subsequent pull request, maybe together with `install_arrow()`.